blurring_as_a_service/inference_pipeline/components/detect_and_blur_sensitive_data.py

```diff
@@ -88,75 +79,247 @@ def detect_and_blur_sensitive_data(
          Each row is a relative path to {customer_name}_input_structured/inference_queue
     results_path:
         Where to store the results.
-    yolo_yaml_path:
-        Where to store the yaml file which is used during validation
     customer_name
         The name of the customer, with spaces replaced by underscores
     model_parameters_json
         All parameters used to run YOLOv5 inference in json format
     database_parameters_json
         Database credentials
-
     """
-    # Check if the folder exists
+    start_time = get_current_time()
     logger = logging.getLogger("detect_and_blur_sensitive_data")
     if not os.path.exists(batches_files_path):
         raise FileNotFoundError(f"The folder '{batches_files_path}' does not exist.")
-    datastore_output_path = settings["inference_pipeline"]["datastore_output_path"]
+    datastore_output_path = settings["inference_pipeline"]["outputs"]["output_rel_path"]
     if datastore_output_path:
         results_path = os.path.join(results_path, datastore_output_path)
-    # Iterate over files in the folder
-    for batch_file_txt in os.listdir(batches_files_path):
+    batch_files_to_iterate = os.listdir(batches_files_path)
+    logging.info(f"Batches file to do: {batch_files_to_iterate}")
+    error_trace = ""
+    for batch_file_txt in batch_files_to_iterate:
         if batch_file_txt.endswith(".txt"):
             file_path = os.path.join(batches_files_path, batch_file_txt)
+            try:
+                if os.path.isfile(file_path) and os.path.exists(file_path):
```

Comment:
`os.path.exists` is redundant if `os.path.isfile` already returns True. So you only need `os.path.isfile` here.